src/screens/schedulesHistoric/hooks/useFiltersForm.ts

```diff
@@ -0,0 +1,181 @@
+import { SelectChangeEvent } from '@mui/material';
+import { Dayjs } from 'dayjs';
+import { useEffect, useState } from 'react';
+import useGetAllProfessors from '../../../service/requests/useGetAllProfessors';
+import useGetProfessorSubjects from '../../../service/requests/useGetProfessorSubjects';
+import { TGetSchedulesHistoricRequestParams } from '../../../service/requests/useGetSchedulesHistoricRequest/types';
+import useGetLoggedUser from '../../../service/storage/getLoggedUser';
+import { TypeUserEnum } from '../../../utils/constants';
+import { useSnackBar } from '../../../utils/renderSnackBar';
+import { TScheduleHistoricFilters } from './types';
+
+type Props = {
+  getSchedules(params: TGetSchedulesHistoricRequestParams): void;
+  setPage(page: React.SetStateAction<number>): void;
+};
+
+const useSchedulesHistoric = ({ getSchedules, setPage }: Props) => {
+  const { showErrorSnackBar } = useSnackBar();
+
+  const user = useGetLoggedUser();
+
+  const {
+    error: errorGetProfessorSubjects,
+    getProfessorSubjects,
+    isLoading: isLoadingGetProfessorSubjects,
+    response: responseGetProfessorSubjects,
+  } = useGetProfessorSubjects();
+
+  const {
+    error: errorGetAllProfessors,
+    getProfessors,
+    isLoading: isLoadingGetAllProfessors,
+    response: responseGetAllProfessors,
+  } = useGetAllProfessors();
+
+  const [filters, setFilters] = useState<TScheduleHistoricFilters>({
+    nameOrEnrollFilter: '',
+    beginDateFilter: null,
+    endDateFilter: null,
+    responsiblesOrSubjectsFilter: [],
+  });
+
+  const handleFiltersChange = (filters: TScheduleHistoricFilters) => {
+    setFilters(filters);
+  };
+
+  const handleChangeBeginDateFilter = (date: Dayjs | null) => {
+    const currentFilters: TScheduleHistoricFilters = {
+      nameOrEnrollFilter: filters.nameOrEnrollFilter,
+      beginDateFilter: date,
+      endDateFilter: filters.endDateFilter,
+      responsiblesOrSubjectsFilter: filters.responsiblesOrSubjectsFilter,
```

Comment:
Para economizar linha de código:
```suggestion
      ...filters,
      beginDateFilter: date,
```

Voce pode replicar isso para todos os handleChange's.

Reply:
Feito